### PR TITLE
[hotfix][table-planner-blink] Hotfix for window join: joinCondition is wrong  if join condition contains 'IS NOT DISTINCT FROM'.

### PR DESCRIPTION
**Pull request by beyond1920**
## What is the purpose of the change
The pr aims to hotfix a bug in pr relates to [Flink-22098](https://issues.apache.org/jira/browse/FLINK-22098) caused by a mistake when rebasing (Sorry about that). 

## Brief change log
  - Fix a bug in WindowJoinUtil
  - Update the wrong planner in WindowJoinTest.xml

## Verifying this change

  - Existed UT in WindowJoinTest*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowJoinUtil.scala [hunk 1]
```diff
@@ -98,31 +98,37 @@ object WindowJoinUtil {
...
+          val remainEqual = rexBuilder.makeCall(op, leftInputRef, rightInputRef)
+          remainEquals += remainEqual
+          remainLeftKeysArray += source
+          remainRightKeysArray += target
         }
       }
-      val remainAnds = remainEquals ++ joinInfo.nonEquiConditions
+      val notEquiCondition = joinSpec.getNonEquiCondition
+      if (notEquiCondition.isPresent) {
+        remainEquals += notEquiCondition.get()
```
Comment (godfreyhe) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowJoinUtil.scala:123]: it's better we can rename this field to `remainingConditions`

**Review by godfreyhe — APPROVED** [review]
LGTM except one minor comment

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowJoinUtil.scala (hunk 1 above)
Comment (beyond1920) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowJoinUtil.scala]: done

**Comment by beyond1920** [comment]
@flinkbot run azure

**Pull request closed without merging**